test_crates/traits_with_types/Cargo.toml

```diff
@@ -0,0 +1,9 @@
+[package]
+publish = false
+name = "traits_with_types"
```

Comment:
nit: `traits_with_types` is a bit ambiguous of a name, it doesn't immediately communicate what we're going for here.

How about naming the crate, its directory, and the test case file `traits_with_associated_types`?

Reply:
Sure, done in e01dadf